src/synapse.py

```diff
@@ -83,6 +91,76 @@ def execute_migrate_config(self, container: ops.Container) -> None:
                 "Migrate config failed, please review your charm configuration"
             )
 
+    def check_server_name(self, container: ops.Container) -> None:
+        """Check server_name.
+
+        Check if server_name of the state has been modified in relation to the configuration file.
+
+        Args:
+            container: Container of the charm.
+
+        Raises:
+            PathError: if somethings goes wrong while reading the configuration file.
+            ServerNameModifiedError: if server_name from state is different than the one in the
+                configuration file.
+        """
+        try:
+            configuration_content = str(
+                container.pull(SYNAPSE_CONFIG_PATH, encoding="utf-8").read()
+            )
+            configured_server_name = yaml.safe_load(configuration_content)["server_name"]
+            if (
+                configured_server_name is not None
+                and configured_server_name != self._charm_state.server_name
+            ):
+                msg = (
+                    f"server_name {self._charm_state.server_name} is different from the existing "
```

Comment:
I think it is worthwhile adding a sentence explaining that the server_name can't be changed and pointing to some Synapse documentation